sematic/retry_settings.py

```diff
@@ -0,0 +1,45 @@
+# Standard Library
+from dataclasses import dataclass, field
+from typing import Tuple, Type
+
+# Sematic
+from sematic.utils.exceptions import ExceptionMetadata
+
+
+@dataclass
+class RetrySettings:
+    """
+    Configuration object to pass to `@sematic.func` to activate retries.
+
+    Parameters
+    ----------
+    exceptions: Tuple[Type[Exception]]
+        A tuple of exception types to retry for.
+
+    times: int
+        How may times to retry.
+    """
+
+    exceptions: Tuple[Type[Exception]]
+    times: int
```

Comment:
Suggest calling this `extra_times` or `n_retries` instead of `times`. Otherwise `times=3` means it will execute potentially 4 times, which is kind of counter-intuitive. But `extra_times=3` meaning 4 executions is more understandable